composition-js/src/merging/merge.ts

```diff
@@ -353,6 +369,7 @@ class Merger {
     // supergraph. This allow to be able to reference those from that point on.
     this.addTypesShallow();
     this.addDirectivesShallow();
+    this.addCustomTypeSystemDirectives();
```

Comment:
I'm running a bit out of time to dig into this method more deeply today so this may turn out to be a stupid remark, but it's not obvious to me a priori why we need that method, and why `addDirectivesShallow` doesn't handle this directly. It certainly was the initial intent that it would.

I'm also not sure why we need to do something specific about the locations at that stages. Isn't that handled by ` mergeTypeSystemDirectiveDefinition`?